tools/appliance/systemvmtemplate/scripts/install_systemvm_packages.sh

```diff
@@ -70,20 +76,33 @@ function install_packages() {
     radvd \
     sharutils genisoimage aria2 \
     strongswan libcharon-extra-plugins libstrongswan-extra-plugins strongswan-charon strongswan-starter \
-    virt-what open-vm-tools qemu-guest-agent hyperv-daemons
+    virt-what open-vm-tools qemu-guest-agent hyperv-daemons cloud-guest-utils \
+    apt-transport-https ca-certificates curl gnupg  gnupg-agent software-properties-common cloud-init
 
   apt-get -y autoremove --purge
   apt-get clean
   apt-get autoclean
 
+  apt_clean
   ${apt_get} install links
 
+   curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add -
+   apt-key fingerprint 0EBFCD88
+
   #32 bit architecture support for vhd-util: not required for 32 bit template
   if [ "${arch}" != "i386" ]; then
     dpkg --add-architecture i386
     apt-get update
     ${apt_get} install libuuid1:i386 libc6:i386
+
+    add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/debian \
+    $(lsb_release -cs) \
+    stable"
+    apt-get update
+    ${apt_get} install docker-ce docker-ce-cli containerd.io
```

Comment:
@davidjumani
we do not need docker installed in systemvm template.
it would be better to create separated folder in tools/appliance/

Reply:
@weizhouapache With coreOS having reached EOL, we've enabled the systemVM template to be consumed by CKS for deploying K8s clusters. Hence docker is installed in the template. However, the service is only run on the cks nodes

Reply:
@Pearl1594 
we could have different template for cks. debian9 is one of an options for cks template. if cloudstack users want to have centos8 or ubuntu 2004 as cks template,we could build them as well.
we can create separated folder in tools/appliance/, like cks-template-debian10, cks-template-centos8, etc

if we use same template on VR and cks, and we need to make more changes in the template, it will impact the systemvm and vrs as well. cloudstack systemvm template is quite stable now. do not touch it if possible.
in addition, some packages in systemvm template are not needed for cks.

Reply:
@weizhouapache I completely understand your concerns. However, the basic intent of moving to systemVM template was that we could leverage the already registered template, thus removing the need to register a new template for CKS. We've ensured that VRs and systemVMs will not be affected by these new packages installed. In case the concern is about the template size, it's increased by 500MB and have incorporated your PR https://github.com/apache/cloudstack/pull/3900 for growing / resizing the root disk partition during boot up to a predefined / provided disk size. We could also probably discuss this on the mailing list to get a feedback from a wider audience and incorporate and test all cases to ensure that there aren't any adverse effects of these changes.. Thanks @weizhouapache

Reply:
@Pearl1594 thanks !
It would be better to discuss it on mailing list.